src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -98,176 +97,206 @@ public static List<String> appendExperiments(List<String> allExperiments, String
+        String uniqueKey = null;
+        try {
+            // Process labels in the 'include' section
+            if (filter != null && filter.getInclude() != null) {
+                // Initialize StringBuilder for uniqueKey
+                StringBuilder includeLabelsBuilder = new StringBuilder();
+                Map<String, String> includeLabels = filter.getInclude().getLabels();
+                if (includeLabels != null && !includeLabels.isEmpty()) {
+                    includeLabels.forEach((key, value) ->
+                            includeLabelsBuilder.append(key).append("=").append("\"" + value + "\"").append(",")
+                    );
+                    // Remove trailing comma
+                    if (includeLabelsBuilder.length() > 0) {
+                        includeLabelsBuilder.setLength(includeLabelsBuilder.length() - 1);
                     }
-                    // Close the connection
-                    connection.disconnect();
-                });
+                    LOGGER.debug("Include Labels: " + includeLabelsBuilder.toString());
+                    uniqueKey = includeLabelsBuilder.toString();
+                }
             }
         } catch (Exception e) {
-            LOGGER.error(e.getMessage());
             e.printStackTrace();
-            jobStatusMap.get(jobID).setStatus("FAILED");
+            LOGGER.error(e.getMessage());
         }
+        return uniqueKey;
     }
+
+    private JSONObject processDateRange(BulkInput.TimeRange timeRange) {
+        JSONObject dateRange = null;
+        if (null != timeRange && timeRange.getStart() != null && timeRange.getEnd() != null) {
+            String intervalEndTimeStr = timeRange.getStart();
+            String intervalStartTimeStr = timeRange.getEnd();
+            long interval_end_time_epoc = 0;
+            long interval_start_time_epoc = 0;
+            LocalDateTime localDateTime = LocalDateTime.parse(intervalEndTimeStr, DateTimeFormatter.ofPattern(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT));
+            interval_end_time_epoc = localDateTime.toEpochSecond(ZoneOffset.UTC);
+            Timestamp interval_end_time = Timestamp.from(localDateTime.toInstant(ZoneOffset.UTC));
+            localDateTime = LocalDateTime.parse(intervalStartTimeStr, DateTimeFormatter.ofPattern(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT));
+            interval_start_time_epoc = localDateTime.toEpochSecond(ZoneOffset.UTC);
+            Timestamp interval_start_time = Timestamp.from(localDateTime.toInstant(ZoneOffset.UTC));
+            int steps = CREATE_EXPERIMENT_CONFIG_BEAN.getMeasurementDuration() * KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE; // todo fetch experiment recommendations setting measurement
+            dateRange = new JSONObject();
+            dateRange.put("start_time", interval_start_time_epoc);
+            dateRange.put("end_time", interval_end_time_epoc);
+            dateRange.put("steps", steps);
```

Comment:
`steps` seems to be misnamed, can we call it `measurementDuration` if that is what it is?